twins/api/http/endpoint_twins_test.go

```diff
@@ -259,58 +271,68 @@ func TestUpdateTwin(t *testing.T) {
 			contentType: contentType,
 			auth:        token,
 			status:      http.StatusNotFound,
+			err:         svcerr.ErrNotFound,
 		},
 		{
 			desc:        "update twin with invalid token",
 			req:         data,
-			id:          stw.ID,
+			id:          twin.ID,
 			contentType: contentType,
 			auth:        authmocks.InvalidValue,
 			status:      http.StatusUnauthorized,
+			err:         svcerr.ErrAuthentication,
 		},
 		{
 			desc:        "update twin with empty token",
 			req:         data,
-			id:          stw.ID,
+			id:          twin.ID,
 			contentType: contentType,
 			auth:        "",
 			status:      http.StatusUnauthorized,
+			err:         svcerr.ErrAuthentication,
 		},
 		{
 			desc:        "update twin with invalid data format",
 			req:         "{",
-			id:          stw.ID,
+			id:          twin.ID,
 			contentType: contentType,
 			auth:        token,
 			status:      http.StatusBadRequest,
+			err:         svcerr.ErrMalformedEntity,
 		},
 		{
 			desc:        "update twin with empty request",
 			req:         "",
-			id:          stw.ID,
+			id:          twin.ID,
 			contentType: contentType,
 			auth:        token,
 			status:      http.StatusBadRequest,
+			err:         svcerr.ErrMalformedEntity,
 		},
 		{
 			desc:        "update twin without content type",
 			req:         data,
-			id:          stw.ID,
+			id:          twin.ID,
 			contentType: "",
 			auth:        token,
 			status:      http.StatusUnsupportedMediaType,
+			err:         apiutil.ErrUnsupportedContentType,
 		},
 		{
 			desc:        "update twin with invalid name",
 			req:         invalidData,
 			contentType: contentType,
 			auth:        token,
 			status:      http.StatusMethodNotAllowed,
+			err:         svcerr.ErrMalformedEntity,
 		},
 	}
 
 	for _, tc := range cases {
 		repoCall := auth.On("Identify", mock.Anything, &magistrala.IdentityReq{Token: tc.auth}).Return(&magistrala.IdentityRes{Id: testsutil.GenerateUUID(t)}, nil)
+		repoCall1 := twinRepo.On("RetrieveByID", mock.Anything, tc.id).Return(twins.Twin{}, tc.err)
```

Comment:
Simulate different failures have error variables for `retrieveByIDErr`, `updateErr` and `cacheErr`